client/rest/test/plugins/metadata/metadataRoutes_spec.js

```diff
@@ -240,4 +240,67 @@ describe('metadata routes', () => {
 			});
 		});
 	});
+
+	describe('metal', () => {
+		const mockServer = new MockServer();
+		const db = {
+			binDataByMetalId: sinon.stub().resolves({ payload: 'db_payload', text: 'db_text' })
+		};
+		metadataRoutes.register(mockServer.server, db, {});
+
+		beforeEach(() => {
+			mockServer.resetStats();
+			mockServer.res.setHeader = sinon.spy();
+			mockServer.res.write = sinon.spy();
+			mockServer.res.end = sinon.spy();
+		});
+
+		describe('get by metal id', () => {
+			const callRouteAndAssert = async (route, req, shouldCallDb) => {
+				await mockServer.callRoute(route, req).then(() => {
+					expect(mockServer.res.write.calledOnce).to.equal(true);
+					expect(mockServer.next.calledOnce).to.equal(true);
+					expect(mockServer.res.setHeader.calledWithExactly('content-type', 'image/png')).to.equal(true);
+					expect(mockServer.res.setHeader
+						.calledWithExactly('Content-Disposition', 'attachment; filename="image.png"')).to.equal(true);
+					expect(mockServer.res.setHeader.calledWithExactly('Content-MetalText', 'db_text')).to.equal(true);
+					expect(mockServer.res.setHeader.calledWithExactly('Content-MetalText', 'db_fail_text')).to.equal(false);
```

Comment:
you are correct that this (second) assert is excessive. the previous one is sufficient, so you can remove this one.

Reply:
👍